test/sim/moves/pursuit.js

```diff
@@ -70,4 +70,22 @@ describe(`Pursuit`, function () {
 		battle.makeChoices('move Pursuit mega -2, switch 3', 'auto');
 		assert.equal(furret.maxhp - furret.hp, 66);
 	});
+
+	it(`should deal damage prior to attacker selecting a switch in after u-turn etc`, function () {
+		battle = common.createBattle([[
+			{species: 'tyranitar', moves: ['pursuit']},
+		], [
+			{species: 'emolga', moves: ['voltswitch']},
+			{species: 'zapdos', moves: ['batonpass']},
+		]]);
+		battle.makeChoices('move Pursuit', 'move voltswitch');
+		assert.false.fullHP(battle.p2.pokemon[0]);
+		battle.choose('p2', 'switch 2');
+		assert.equal(battle.p2.pokemon[0].name, "Zapdos");
+		battle.makeChoices('move Pursuit', 'move batonpass');
+		battle.choose('p2', 'switch 2');
+		assert.fullHP(battle.p2.pokemon[1], 'should not hit Pokemon that has used Baton Pass');
```

Comment:
I imagine it's possible for Emolga to faint to the first Pursuit should it happen to be a critical hit, which will make this check fail. I'd suggest using a weaker attacker and/or a less frail defender.

Reply:
Swapped ttar out for parasect.